src/ray/gcs/gcs_server/gcs_placement_group_manager.cc

```diff
@@ -147,6 +148,20 @@ void GcsPlacementGroupManager::RegisterPlacementGroup(
     }
     return;
   }
+  if (!placement_group->GetName().empty()) {
+    auto it = named_placement_groups_.find(placement_group->GetName());
+    if (it == named_placement_groups_.end()) {
+      named_placement_groups_.emplace(placement_group->GetName(),
+                                      placement_group->GetPlacementGroupID());
+    } else {
+      std::stringstream stream;
+      stream << "Placement Group with name '" << placement_group->GetName()
+             << "' already exists.";
+      RAY_LOG(WARNING) << stream.str();
+      callback(Status::Invalid(stream.str()));
```

Comment:
Return STATUS_CODE_NOT_FOUND instead of Invalid?

Reply:
hm...  I don't think so, we should return NOT_FOUND status code in `getxxx` case, but this is creation case.